### ISSUE TITLE
HFS+: Failure to analyze Capture One 23 disk image

### ISSUE THREAD

**Issue by boredzo** [boredzo/impluse-hfs#22]
```
% hdiutil attach -nomount -readonly …/CaptureOne23.Mac.16.0.1.17.dmg
⋮
verified   CRC32 $74187236
/dev/disk7          	GUID_partition_scheme          	
/dev/disk7s1        	Apple_HFS                      	
% file - < /dev/disk7s1
/dev/stdin: Macintosh HFS Extended version 4 data last mounted by: '10.0', created: Mon Nov 21 03:46:16 2022, last modified: Mon Nov 21 10:46:34 2022, last checked: Mon Nov 21 10:46:16 2022, block size: 4096, number of blocks: 467222, free blocks: 206980
% xxd < /dev/disk7s1 | grep -E '^00000400'
00000400: 482b 0004 8000 0100 3130 2e30 0000 0000  H+......10.0....
% impluse-hfs  analyze /dev/disk7s1
2023-03-07 19:48:48.182 impluse-hfs[20086:30199945] Failed: Catalog file was invalid, corrupt, or not where the volume header said it would be
```

**Comment by boredzo** [comment]
fsck_hfs has no problem with the volume:

```
** /dev/rdisk30s1 (NO WRITE)
   Executing fsck_hfs (version hfs-583.100.10).
** Checking non-journaled HFS Plus Volume.
   The volume name is Capture One 23
** Checking extents overflow file.
** Checking catalog file.
** Checking multi-linked files.
** Checking catalog hierarchy.
** Checking extended attributes file.
** Checking volume bitmap.
** Checking volume information.
** The volume Capture One 23 appears to be OK.
```

**Comment by boredzo** [comment]
Stepping through the debugger, analyze is finding the first node in the catalog file to be a leaf node. 🤨

Unclear whether this is a valid leaf node or not. Could be misaligned read (in the middle of a node) or a misplaced read (too late in the file, missing the header node, or possibly outside of the file entirely).

**Comment by boredzo** [comment]
If I view the volume in Synalyze It!, it says the catalog file starts at block 208. If I jump to block 208 (position 4096*208), that looks like a header node.

Need to figure out what offset impluse is trying to read from.

**Comment by boredzo** [comment]
```
Reading extent #0: start block #208, length 704 blocks
Reading extent #1: start block #172,511, length 704 blocks
```

Hmmm. So that much is right. Another buffer-overwriting bug?

**Comment by boredzo** [comment]
Ah, yup. This is the same bug that was fixed in HFS volumes by [ebd494a3].
